### PR TITLE
Add Kong gateway guide

### PR DESCRIPTION
**Pull request by sixhobbits**

### CODE REVIEW

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+Navigate to your FusionAuth instance.
+
+First, you need to make sure the JWT issuer setting is correct. Navigate to [breadcrumb]#Tenants -> Your Tenant# and change the [field]#Issuer# field to the URL of your FusionAuth instance, for example, `\https://local.fusionauth.io`. Record this value, because we will use it later when generating the JWT.
+
+image::api-gateways/kong-gateway/kong-gateway-tenant-issuer.png[Tenant issuer.,width=1200px]
+
+Next, you need to configure an application that will issue tokens to access the Kong Gateway service.
+
+Navigate to [breadcrumb]#Applications# and create a new application. Fill out the [field]#Name# field, then click the [breadcrumb]#JWT# tab and toggle the [field]#Enabled# switch. Select `Auto generate a new key on save...` for the [field]#Access Token signing key# field. This will generate an RS256 asymmetric key pair specifically for this application.
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:60]: should be `[uielement]#Auto generate a new key on save...#`

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+image::api-gateways/kong-gateway/kong-gateway-fusionauth-jwt-settings.png[FusionAuth application JWT settings,width=1200]
+
+Click the [uielement]#Save# button.
+
+Now, navigate to [breadcrumb]#Settings -> Key Master#. You will see the access key that you've just created in this list. By default, it has the name `Access token signing key generated for application <NAME_OF_APPLICATION>`. View the key and copy the entire contents of the [field]#PEM encoded# field under the [breadcrumb]#Public Key# section into a file in your current working directory. Name the file something like `public.pem`
+
+image::api-gateways/kong-gateway/kong-gateway-public-key.png[FusionAuth Public Key for Access Token,width=1200]
+
+You will use FusionAuth's API to generate a test JWT for your application. To do this, you need an API key with POST access to the `/api/login` endpoint. Go to [breadcrumb]#Settings -> API Keys# to add the key. Make sure the appropriate tenant is selected in the [field]#Tenant# field.
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:70]:
```suggestion
You will use FusionAuth's API to generate a test JWT for your application. However, in a real world application, the user or service would get the JWT through an OAuth grant. In the case of a user, the appropriate grant is the Authorization Code grant, and for a service, the Client Credentials grant. To make this example simpler, you'll use the Login API to get the JWT. To do this, you need an API key with POST access to the `/api/login` endpoint. Go to [breadcrumb]#Settings -> API Keys# to add the key. Make sure the appropriate tenant is selected in the [field]#Tenant# field.
```

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+Execute the following command in your terminal to add a consumer to Kong.
+
+[source,shell,title="Creating a consumer"]
+----
+curl -i -X POST http://localhost:8001/consumers \
+  --data "username=fusionauth"
+----
+
+This command creates a Kong consumer with the username `fusionauth`. This consumer will be linked to the authentication information that you supply to it, namely the [field]#Public key# and [field]#Issuer# name from FusionAuth. This consumer uses the information provided to verify any JWT used to access the service. You can enable this with the following command:
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:94]: Can you use JWKS to have kong get the public key based on the JWT and issuer or do you have to provide it up front as you do?

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+--data-raw '  {
+    "loginId": "<USER_EMAIL>",
+    "password": "<USER_PASSWORD>",
+    "applicationId": "<APPLICATION_ID>",
+    "noJWT" : false
+  }'
+----
+
+Here, `YOUR_FUSIONAUTH_URL` is the [field]#Issuer# name, and `API_KEY` is the key you set up on the [breadcrumb]#Settings -> API Keys# page.
+
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:134]: A bit weird to have the picture here, as the user isn't on that screen any more. please move this picture up to where you create the API key.

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+image::api-gateways/kong-gateway/kong-gateway-api-key.png[FusionAuth API Key setup, width=1200,role=bottom-cropped]
+
+For `APPLICATION_ID`, use the Id of your FusionAuth application.
+
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:138]: Same thing here. either need to tell the person how to get to that screen or capture it when you create the application.

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+image::api-gateways/kong-gateway/kong-gateway-application-id.png[FusionAuth Application Id, width=1200,role=bottom-cropped]
+
+The values for `USER_EMAIL` and `USER_PASSWORD` are the username and password of the test user that you registered to that application.
+
+Using the example names from this guide, the command will appear as follows:
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:143]:
```suggestion
Using the example values from this guide, the command would be:
```

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+[source,json,title="Token response from API call"]
+----
+{
+    "token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImhDUjA4X3daR2s0OUFlYUFmRDY5ZmJKWmRGTSJ9.eyJhdWQiOiI2M2I3M2Y3Ni03NDAwLTQ4N2QtYjEyMi01NzA1Yjg0OGRhODAiLCJleHAiOjE2NzMzNjYyMDQsImlhdCI6MTY3MzM2MjYwNCwiaXNzIjoiaHR0cHM6Ly9mdXNpb25hdXRoLnJpdHphLmNvIiwic3ViIjoiMzk2MzAwMGYtNjg2ZC00MTY5LWI2MjgtOWM5YzQ1MzRiNzgwIiwianRpIjoiZDk3ZGIyZWYtZjExNS00ZDIxLWFlOTQtMDIyN2RmMGU4YzI5IiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6ImJvYkBhd3MuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInByZWZlcnJlZF91c2VybmFtZSI6ImJvYmF3cyIsImFwcGxpY2F0aW9uSWQiOiI2M2I3M2Y3Ni03NDAwLTQ4N2QtYjEyMi01NzA1Yjg0OGRhODAiLCJyb2xlcyI6W10sImF1dGhfdGltZSI6MTY3MzM2MjYwNCwidGlkIjoiZjAwNGMxZmUtNDg0Yi05MDJjLWQ3Y2EtYmRiYzQ2NGRhMGI3In0.m7gzXhNLToPNVE1p5Vo2pLgP6WBcPNfS_zZJnJ81mdEgi6-orViz-tU8j0L8wva0-8KlMdy54cq_XjnDnYJ0aX90O4ZE_QVU5NuDDfzXH14wQtKQoIIydsB6ZvQoBt8JNFUHJb9ANLCGnfn6FVQKqPIzye18Gx_7wYSVokw3eLNFyzrq9dwOD5Q8V9gvZmXV2pTokQAtA7qFaadb2dIeFlSEB7wamKiZLXILjeWAeMbbvAAMQZWFh46UJjwr06QTd8PxQmRwDWWznJy1Vs8EAgZA4vkRSWnn3IbiaCtOaL1ANuEex6il7q32ahxj0Ncm9wn0DbDsQE9NB0CCNTSIhA",
+    "tokenExpirationInstant": 1673366204805,
+    "user": {
+        ....
```
Comment (mooreds) [inline site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:166]:
```suggestion
      "sampleuserdata" : "..."
```

This is valid json and looks better.

**Review by mooreds — CHANGES_REQUESTED** [review]
did you end up writing a kong oidc guide?

In general don't use "we", use "you". Please update to use "you".

Add same troubleshooting section as we had with AWS API gateway, with at least the token expiring. And any other ones.

I like the next steps thing, we should extract that to a fragment and add more tutorials there.

File: site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc (hunk 3 above)
Comment (TheMiniDriver) [reply site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc:94]: As far as I can see in the docs (https://docs.konghq.com/hub/kong-inc/jwt/), it doesn't have support for jwks. Their Auth0 example also sets the public key as we have as well.

Comment (mooreds) [reply site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc]: Makes sense. Seems like they reserve that for the OIDC plugin: https://github.com/Kong/kong/issues/8381#issuecomment-1035441828

**Pull request merged**